### ISSUE TITLE
508-defect-2 [AXE-CORE]: Zoom In / Zoom Out controls MUST have their visible text as part of their accessible name

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#6104]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
! Leaflet-Mapbox issue
```

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The Zoom In / Zoom Out controls **must** have their visible text as part of their accessible name.
The complete visible text content of the target element either matches or is contained within its accessible name.

Speech input users can interact with a webpage by speaking the visible text labels of menus, links, and buttons that appear on the screen. It is confusing to speech input users when they say a visible text label they see, but the speech command does not work because the component's accessible (programmatic) name does not match the visible label. When a user interface component has a visible text label — whether it be real text or an image of text — that text must also be found in the component's accessible (programmatic) name. When the visible label and accessible (programmatic) name for interactive components are in sync, speech input users can effectively interact with those components.

```diff
- "Must" for if the feedback must be applied 
! "Should" if the feedback is best practice 
+ "Consider" for suggestions/enhancements 
```

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a speech input user, I want to interact with the map controls by speaking the visible text labels.

## Possible Fixes

Fix this issue by ensuring that the accessible name — such as the `aria-label` — contains the text of the visible label in the same order in which it appears. While it is not necessary that the visible label and accessible name match exactly, it is a best practice to have the accessible name begin with the visible text.

### Current Code
```html
<a class="leaflet-control-zoom-in" href="#" title="Zoom in" 
role="button" aria-label="Zoom in">+</a>
```

### Recommended Code
```html
<a class="leaflet-control-zoom-in" href="#" title="Zoom in" 
  role="button" aria-label="Zoom in">Zoom in</a>
```

## WCAG or Vendor Guidance (optional)

* [axe-core 3.4 - Label and name from content mismatch](https://dequeuniversity.com/rules/axe/3.4/label-content-name-mismatch?application=coconut)

**Comment by mmiddaugh** [comment]
@jenstrickland is it necessary to announce "Zoom in/out" when the buttons themselves (+/-) seem to be universally understood for those functions on maps and in other applications, such as viewing documents?

**Comment by jenstrickland** [comment]
@mmiddaugh For screen readers and people who use speech input devices the plus and minus signs don't provide the cues needed to trigger the action. A speech input user would say "Plus" or "Minus" but it wouldn't trigger the action.

**Comment by davidconlon** [comment]
@jenstrickland I strongly disagree that this is a level 1. 

I agree with @mmiddaugh 's comment above that this is the standard, but more importantly this is also not a function _any_ map service allows us to modify.
As a launch blocker you are saying that we need to change technologies to something that, at cursory analysis of the market, does not exist.

Would you like to meet and discuss?

**Comment by jenstrickland** [comment]
I was just in a meeting with @mmiddaugh and @brangi. They showed me how the map is included. There is no way for this team to address the accessibility issues. Therefore I will adjust this to a 508-defect-2, and we will work with Mapbox to address the issues (as possible). Frankly, I investigated other maps on the market, and they simply block the map from assistive tech. The VA 508 Office is okay with this, if we must. It creates confusion for low-vision users, but I suspect it's something they, sadly, experience regularly.

**Comment by mmiddaugh** [comment]
Per conversation with @joshkimux on 1/11 - Zoom In/zoom Out does not need to be visually displayed as the controls are announced. Work will be done to announce contextual changes and improve the keyboard navigation and focus in separate issues.

**Comment by joshkimux** [comment]
@mmiddaugh I'm ok with closing this out-- but would it be possible for us to flag this as something to test with mobile screen reader users this year?
